scripts/zkai_d64_gate_value_projection_proof_input.py

```
@@ -0,0 +1,496 @@
+def require_signed_m31(value: int, label: str) -> None:
+    if value <= -M31_MODULUS or value >= M31_MODULUS:
+        raise GateValueProjectionInputError(f"{label} outside signed M31 bounds")
+
+
+def load_bridge(path: pathlib.Path = BRIDGE_JSON) -> dict[str, Any]:
+    try:
+        source_size = path.stat().st_size
+        if source_size > MAX_BRIDGE_JSON_BYTES:
+            raise GateValueProjectionInputError(
+                f"source bridge evidence exceeds max size: got {source_size} bytes, limit {MAX_BRIDGE_JSON_BYTES} bytes"
+            )
```

Ruff (0.15.12)

[warning] Avoid specifying long messages outside the exception class

(TRY003)